bindings/python/pire.pyx.mako

```diff
@@ -0,0 +1,369 @@
+                )
+            self.scanner_impl = impl.CountingScanner(pattern.fsm_impl, sep.fsm_impl)
+    % endif
+
+    def Save(self):
+        cdef impl.yostream stream
+        self.scanner_impl.Save(&stream)
+        return stream.GetStr()
+
+    Load = _${Scanner}_Load
+
+    def InitState(self):
+        return ${Scanner}State.__new__(${Scanner}State, self)
+
+    % if "Glue" not in spec.ignored_methods:
+    def GluedWith(self, ${Scanner} rhs not None, size_t max_size=0):
+        return wrap_${Scanner}(impl.Glue(self.scanner_impl, rhs.scanner_impl, max_size))
+    % endif
+
+    % for method in ["Size", "Empty", "RegexpsCount", "LettersCount"]:
+    %     if method not in spec.ignored_methods:
+    def ${method}(self):
+        return self.scanner_impl.${method}()
+    %     endif
+    % endfor
+
+    def Matches(self, bytes line not None):
+        return impl.Matches(self. scanner_impl, begin(line), end(line))
+
+    % for method in ["LongestPrefix", "ShortestPrefix"]:
+    def ${method}(self, bytes line not None):
+        cdef:
+            const char* line_begin = begin(line)
+            const char* line_end = end(line)
+            const char* prefix_end
+        prefix_end = impl.${method}(self.scanner_impl, line_begin, line_end)
+        if prefix_end != NULL:
+            return prefix_end - line_begin
+        return None
+    % endfor
+
+    % for method in ["LongestSuffix", "ShortestSuffix"]:
+    def ${method}(self, bytes line not None):
+        cdef:
+            const char* rbegin = end(line) - 1
+            const char* rend = begin(line) - 1
+            const char* suffix_begin
+        suffix_begin = impl.${method}(self.scanner_impl, rbegin, rend)
+        if suffix_begin != NULL:
+            return suffix_begin - rend
+        return None
+    % endfor
+
+    StateType = ${Scanner}State
+% endfor
+
+
+
+cdef class Options:
+    cdef set option_set
+
+    def __cinit__(self, set option_set=None):
+        if option_set is None:
+            option_set = set()
+        self.option_set = option_set
+
+    def __init__(self):
+        pass
+
+    @staticmethod
+    def Parse(bytes letters not None):
+        cdef parsed = Options()
+        for letter in letters:
+            parsed |= _LETTER_MAP[letter]
+        return parsed
+
+    def __ior__(self, Options rhs not None):
+        self.option_set |= rhs.option_set
+        return self
+
+    def __or__(Options self not None, Options rhs not None):
+        return Options.__new__(Options, self.option_set | rhs.option_set)
+
+    cdef inline impl.yauto_ptr[impl.Options] Convert(self):
+        return impl.ConvertFlagSetToOptions(<impl.FlagSet>self.option_set)
+
+    cdef inline void Apply(self, Lexer lexer):
+        cdef impl.yauto_ptr[impl.Options] converted = self.Convert()
+        converted.get().Apply(cython.operator.dereference(lexer.lexer_impl))
+
+
+% for option in OPTIONS:
+${option} = Options.__new__(Options, {impl.${option}})
```

Comment:
What `{` and `}` mean in `{impl.${option}}`? If they create a set, replace with `set([...])`, please.